compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -787,6 +797,17 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 );
                 (generics, kind, ty.is_some())
             }
+            AssocItemKind::Delegation(box delegation) => {
+                let delegation_results = self.lower_delegation(delegation);
+                (
+                    delegation_results.generics,
+                    hir::TraitItemKind::Fn(
+                        delegation_results.sig,
+                        hir::TraitFn::Provided(delegation_results.body_id),
```

Comment:
Style nit: these multi-line muti-level tuples look horrible.
Please, use separate statements or shorter names to avoid.